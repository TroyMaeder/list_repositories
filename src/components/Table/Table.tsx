import { FC } from "react";
import { default as MUITable } from "@mui/material/Table";
import { TableBody } from "@mui/material";
import { styled } from "@mui/system";
import { TableContainer } from "@mui/material";
import { TableRow } from "@mui/material";
import { createRow, Row } from "../../helpers/createRow";
import LoadingBoundary from "../LoadingBoundary/LoadingBoundary";

interface Props {
  tableData: Row[] | undefined;
  user?: string;
  loading: boolean;
}

const TableCell = styled("td")({
  width: "50px",
  display: "table-cell",
  textAlign: "center",
  padding: "16px 4px",
});

const Table: FC<Props> = ({ tableData, user, loading }) => {
  const rows = tableData?.map((row) => createRow(row));

  return (
    <LoadingBoundary loading={loading}>
      <TableContainer>
        <MUITable>
          <TableBody>
            {rows?.map((row) => (
              <TableRow key={row.name}>
                <TableCell aria-hidden="true"> - </TableCell>
                <TableCell>
                  <a
                    style={{ textDecoration: "none" }}
                    href={`https://github.com/${user}/${row.name}`}
                  >
                    {row.name}
                  </a>
                </TableCell>
                <TableCell aria-hidden="true"> - </TableCell>
                <TableCell role="img" aria-label="star">
                  {row.starIcon}
                </TableCell>
                <TableCell>{row.stargazerCount}</TableCell>
                <TableCell aria-hidden="true"> - </TableCell>
                <TableCell role="img" aria-label="fork">
                  {row.forkIcon}
                </TableCell>
                <TableCell>{row.forkCount}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </MUITable>
      </TableContainer>
    </LoadingBoundary>
  );
};

export default Table;
