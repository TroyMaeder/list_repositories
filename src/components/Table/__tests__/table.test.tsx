import { render } from "@testing-library/react";
import Table from "../Table";

const tableDataMock = {
  name: "readmes-for-founders-and-coders",
  starIcon: "",
  stargazerCount: 100,
  forkIcon: "",
  forkCount: 200,
};

describe("Table", () => {
  const TableComponent = <Table loading={false} tableData={[tableDataMock]} />;
  render(TableComponent);

  it("should render correct markup", () => {
    const { container } = render(TableComponent);
    expect(container).toMatchSnapshot();
  });
});
