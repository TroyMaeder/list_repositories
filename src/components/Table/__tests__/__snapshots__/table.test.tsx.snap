// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render correct markup 1`] = `
<div>
  <div
    class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            aria-hidden="true"
            class="css-1dqonpg"
          >
             - 
          </td>
          <td
            class="css-1dqonpg"
          >
            <a
              href="https://github.com/undefined/readmes-for-founders-and-coders"
              style="text-decoration: none;"
            >
              readmes-for-founders-and-coders
            </a>
          </td>
          <td
            aria-hidden="true"
            class="css-1dqonpg"
          >
             - 
          </td>
          <td
            aria-label="star"
            class="css-1dqonpg"
            role="img"
          >
            🌟
          </td>
          <td
            class="css-1dqonpg"
          >
            100
          </td>
          <td
            aria-hidden="true"
            class="css-1dqonpg"
          >
             - 
          </td>
          <td
            aria-label="fork"
            class="css-1dqonpg"
            role="img"
          >
            🍴
          </td>
          <td
            class="css-1dqonpg"
          >
            200
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
